### PR TITLE
**Merge addRestaurant-JSP initial build first** Add forms to addRestaurant jsp

### PR DESCRIPTION
**Pull request by angapinski**

### CODE REVIEW

File: WebContent/WOH-addRestaurant.jsp [hunk 1]
```diff
@@ -40,6 +39,70 @@
     	</ul>
   	</div>
 	</nav>
+	
+	<div class ="container">
+	<div class="row">
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: add a tab.  this shows that container is row's parent.

File: WebContent/WOH-addRestaurant.jsp [hunk 2]
```diff
@@ -40,6 +39,70 @@
...
+			<form action = "" method = "POST" class="form-horizontal" id="">
+					
+				
+				
+				<!-- Restaurant Name Textbox -->
+					<div class="form-group">
+					<label class="control-label col-sm-2" for="item">Restaurant Name</label>
+						<div class="col-sm-10">
+         					<input type = "text" class="form-control" name = "restaurantName">
+         				</div>
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: This closing tag needs to match up with the position of the opening tag.  This drastically improves code readability

File: WebContent/WOH-addRestaurant.jsp [hunk 3]
```diff
@@ -40,6 +39,70 @@
...
+         					<input type = "text" class="form-control" name = "distance">
+         				</div>
+         			</div>
+         			
+         		<!-- Food Type Dropdown -->
+					<div class="form-group">
+  					<label class="control-label col-sm-2" for="foodType">Select Food Type</label>
+  						<div class="col-sm-10">
+  							<select class="form-control" name="foodType">
+    							<option value = "Fast-Food" >Fast Food</option>
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: value should be written as fastFood.

File: WebContent/WOH-addRestaurant.jsp [hunk 4]
```diff
@@ -40,6 +39,70 @@
...
+						<div class="col-sm-10">
+         					<input type = "text" class="form-control" name = "restaurantName">
+         				</div>
+         			</div>
+         			
+				<!-- Average Price Textbox -->
+					<div class="form-group">
+					<label class="control-label col-sm-2" for="averagePrice">Price Per Meal</label>
+						<div class="col-sm-10">
+         					<input type = "text" class="form-control" name = "averagePrice">
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: rename averagePrice to price.

File: WebContent/WOH-addRestaurant.jsp [hunk 5]
```diff
@@ -40,6 +39,70 @@
...
+					<div class="form-group">
+					<label class="control-label col-sm-2" for="item">Restaurant Name</label>
+						<div class="col-sm-10">
+         					<input type = "text" class="form-control" name = "restaurantName">
+         				</div>
+         			</div>
+         			
+				<!-- Average Price Textbox -->
+					<div class="form-group">
+					<label class="control-label col-sm-2" for="averagePrice">Price Per Meal</label>
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: if you want to use the "for" attribute, it has to refer to an id or a class.

File: WebContent/WOH-addRestaurant.jsp [hunk 6]
```diff
@@ -40,6 +39,70 @@
...
+         			</div>
+         			
+         		<!-- Food Type Dropdown -->
+					<div class="form-group">
+  					<label class="control-label col-sm-2" for="foodType">Select Food Type</label>
+  						<div class="col-sm-10">
+  							<select class="form-control" name="foodType">
+    							<option value = "Fast-Food" >Fast Food</option>
+    							<option value = "Sit-Down" >Sit-Down</option>
+    							<option value = "Drive-Through" >Drive-Through</option>
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: Recommend changing Drive-Through to Dessert.

File: WebContent/WOH-addRestaurant.jsp [hunk 7]
```diff
@@ -40,6 +39,70 @@
...
+  							</select>
+  						</div>
+  					</div>
+  					
+  					<!-- Cuisine Type Checkbox -->
+  					<div class="form-group">
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: I would recommend setting a specific height to this div and adding "overflow-y: scroll".  This will allow us to add a whole lot more cuisine options in a neat little package.

**Review by aperdew — CHANGES_REQUESTED** [review]

File: WebContent/WOH-addRestaurant.jsp [hunk 8]
```diff
@@ -40,6 +39,70 @@
...
+  						<label><input type="checkbox" value="">American</label>
+					</div>
+					<div class="checkbox">
+  						<label><input type="checkbox" value="">Chinese</label>
+					</div>
+					<div class="checkbox">
+  						<label><input type="checkbox" value="">Mexican</label>
+					</div>
+					</div>
+
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: Also, a text area for a property called "notes" so users can add notes to a restaurant.

File: WebContent/WOH-addRestaurant.jsp [hunk 9]
```diff
@@ -40,6 +39,70 @@
...
+    							<option value = "Fast-Food" >Fast Food</option>
+    							<option value = "Sit-Down" >Sit-Down</option>
+    							<option value = "Drive-Through" >Drive-Through</option>
+  							</select>
+  						</div>
+  					</div>
+  					
+  					<!-- Cuisine Type Checkbox -->
+  					<div class="form-group"> 
+  					<div class="checkbox">
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: for these checkboxes, were going to need to do some more complex stuff to make sure they are properly read and submitted.  Time for JavaScript! Yay!  I'll work with you to implement the function needed to do this.

File: WebContent/WOH-addRestaurant.jsp [hunk 10]
```diff
@@ -40,6 +39,70 @@
...
+  						<label><input type="checkbox" value="">American</label>
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: for each of these checkboxes, set the value to the lowercase version of it's cuisine.  Ex: "american" , "mexican".

Later, we will actually pull these cuisines from a database and use a <c:foreach> to dynamically build these checkboxes.

File: WebContent/WOH-addRestaurant.jsp [hunk 11]
```diff
@@ -40,6 +39,70 @@
...
+					<div class="checkbox">
+  						<label><input type="checkbox" value="">Chinese</label>
+					</div>
+					<div class="checkbox">
+  						<label><input type="checkbox" value="">Mexican</label>
+					</div>
+					</div>
+  				
+  				<!-- Submit Button -->
+  				<input type = "submit" class="btn btn-default" value = "Add Restaurant" />
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: We're going to change how this button behaves when you click it.  We need it to run a javascript function before it submits.  We're going to change it to:
`<button type="button" class="btn btn-default" onclick="formatData()>Submit<button>`
onclick is an attribute that tells the element what to do when it is clicked.  formatData() is the name of the JavaScript function that we're going to implement.

File: WebContent/addRestaurant/addRestaurant.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
+var cuisines ="";
```
Comment (aperdew) [inline WebContent/addRestaurant/addRestaurant.js:1]: So we will need to do something similar for the notes section.  If you can think of any other pieces of information that need to be formatted before submission, you can process them here.

File: WebContent/WOH-addRestaurant.jsp [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+	<title>Wheel of Hunger - Add Restaurant</title>
+</head>
+
+<body>
+<%@include file="banner/WOH-banner.html" %>
+	
+	<div class ="container">
+		<div class="row">
+			<div class="col-sm-10 col-sm-offset-1">
+				<form action = "" method = "POST" class="form-horizontal WOH-addRestaurant-form" id="">
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: remove id from this form

File: WebContent/WOH-addRestaurant.jsp [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+						<div class="form-group">
+						<label class="control-label col-sm-2" for="WOH-addRestaurant-distanceInput">Distance</label>
+							<div class="col-sm-10">
+	         					<input type = "text" class="form-control WOH-addRestaurant-distanceInput" name = "distance">
+	         				</div>
+	         			</div>
+	         			
+	         		<!-- Food Type Dropdown -->
+						<div class="form-group">
+	  					<label class="control-label col-sm-2" for="WOH-addRestaurant-typeInput">Select Food Type</label>
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: Wrap this label in a div with the class col-sm-2 and remove this class from the label

File: WebContent/WOH-addRestaurant.jsp [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+	  						<div class="col-sm-10">
+	  							<select class="form-control WOH-addRestaurants-typeInput" name="type">
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: Were going to need to do some processing of data with this select.

File: WebContent/WOH-restaurants.jsp [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+	<!-- jQuery library -->
+	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+
+	<!-- Latest compiled JavaScript -->
+	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
+	
+	<link rel="stylesheet" type="text/css" href="restaurants/WOH-restaurants.css">
+	<link rel="stylesheet" type="text/css" href="site/WOH-site.css">
+	
+	<title>Wheel of Hunger - Restaurants</title>
```
Comment (aperdew) [inline WebContent/WOH-restaurants.jsp]: Simplify this to just "Restaurants"

File: WebContent/WOH-addRestaurant.jsp [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+
+	<!-- Latest compiled JavaScript -->
+	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
+	<script src="addRestaurant/addRestaurant.js"></script>
+	
+	<link rel="stylesheet" type="text/css" href="addRestaurant/WOH-addRestaurant.css">
+	<link rel="stylesheet" type="text/css" href="site/WOH-site.css">
+	
+	<title>Wheel of Hunger - Add Restaurant</title>
```
Comment (aperdew) [inline WebContent/WOH-addRestaurant.jsp]: Simplify to "Add Restaurant"

File: WebContent/WOH-spinSetup.jsp [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+
+	<!-- Latest compiled JavaScript -->
+	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js"></script>
+	<script src="addRestaurant/addRestaurant.js"></script>
+	
+	<link rel="stylesheet" type="text/css" href="addRestaurant/WOH-addRestaurant.css">
+	<link rel="stylesheet" type="text/css" href="site/WOH-site.css">
+	
+	<title>Wheel of Hunger - Spin Setup</title>
```
Comment (aperdew) [inline WebContent/WOH-spinSetup.jsp]: Simplify to "Spin Setup"

**Review by aperdew — CHANGES_REQUESTED** [review]
